src/loaders/svg/tvgSvgLoader.cpp

```diff
@@ -1343,6 +1357,125 @@ static bool _attrParseSymbolNode(void* data, const char* key, const char* value)
 }
 
 
+template<typename NodeType>
+struct FilterTag
+{
+    const char* tag;
+    SvgParserLengthType type;
+    int sz;
+    size_t offset;
+
+    constexpr FilterTag(const char* tag, SvgParserLengthType type, int size, size_t off)
+        : tag(tag), type(type), sz(size), offset(off) {}
+};
+
+
+template<typename NodeType>
+constexpr FilterTag<NodeType> filterTags[] = {
+    {"x", SvgParserLengthType::Horizontal, sizeof("x"), offsetof(NodeType, x)},
+    {"y", SvgParserLengthType::Vertical, sizeof("y"), offsetof(NodeType, y)},
+    {"width", SvgParserLengthType::Horizontal, sizeof("width"), offsetof(NodeType, width)},
+    {"height", SvgParserLengthType::Vertical, sizeof("height"), offsetof(NodeType, height)}
+};
+
+
+#define PARSE_FILTER_REGION(key, value, loader, NodeType, node) { \
+    unsigned char* array = (unsigned char*)(node); \
+    int sz = strlen(key); \
+    for (unsigned int i = 0; i < sizeof(filterTags<NodeType>) / sizeof(filterTags<NodeType>[0]); i++) { \
+        if (filterTags<NodeType>[i].sz - 1 == sz && !strncmp(filterTags<NodeType>[i].tag, key, sz)) { \
+            *((float*)(array + filterTags<NodeType>[i].offset)) = _toFloat((loader)->svgParse, value, filterTags<NodeType>[i].type); \
+            return true; \
+        } \
+    } \
+}
+
+static bool _attrParseFilterNode(void* data, const char* key, const char* value)
+{
+    SvgLoaderData* loader = (SvgLoaderData*)data;
+    SvgNode* node = loader->svgParse->node;
+    SvgFilterNode* filter = &(node->node.filter);
+
+    PARSE_FILTER_REGION(key, value, loader, SvgFilterNode, filter);
+
+    if (!strcmp(key, "id")) {
+        if (node->id && value) free(node->id);
+        node->id = _copyId(value);
+    } else if (!strcmp(key, "filterUnits")) {
+        if (!strcmp(value, "userSpaceOnUse")) filter->userSpaceFilter = true;
+    } else if (!strcmp(key, "primitiveUnits")) {
+        if (!strcmp(value, "objectBoundingBox")) filter->userSpaceFilter = false;
```

Comment:
Is it okay to use the same variable for filterUnits and [primitiveUnits](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/primitiveUnits)? Both attributes can have the same format of value, but it seems like they are applied to different targets.

If `filterUnits = "userSpaceOnUse", primitiveUnits = "objectBoundingBox"` is set, the user space of the filter is changed to objectBoundingBox.

Reply:
Oh I saw that there is a userSpacePrimitive. This looks a typo.
But additionally, it seems that these values ​​are not actually applied.
Wouldn't it be better to not parse it so that it outputs that it is an unsupported attribute?
(Or how about comment it out?)

Reply:
there is few attribs that are not applied, will remove them